hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -266,7 +315,35 @@ commitTx networkId scriptRegistry headId party utxoToCommitWitnessed (initialInp
   commitDatum =
     mkTxOutDatumInline $ mkCommitDatum party utxoToCommit (headIdToCurrencySymbol headId)
 
-  utxoToCommit = fst <$> utxoToCommitWitnessed
+  TxMetadata metadataMap = mkHydraHeadV1TxName "CommitTx"
+
+  txAuxMetadata = mkAlonzoTxAuxData @[] @LedgerEra (toShelleyMetadata metadataMap) [toLedgerScript initialScript]
+
+-- Re-Associate the redeemer indices present in some tx with the same inputs but different
+-- indices in the current tx.
+--
+-- We need this in order to be able to find the the _old_ tx redeemer and re-add
+-- it to the tx witnesses using the appropriate index of the _new_ tx input
+-- because of the 'TxIn' re-ordering.
+reAssociateOldInputs ::
+  Set (Ledger.TxIn StandardCrypto) ->
+  Set TxIn ->
```

Comment:
Why two different Txfin types here?

Reply:
Well, we do deal with both, cardano-api ones and the ledger ones in this function. I could convert before calling this function if that's what you mean?

Reply:
Consistency between these TxIn types in our internal logic is better I think.